Python/Product/PythonTools/visualstudio_py_debugger.py

```diff
@@ -2155,12 +2152,33 @@ def attach_process(port_num, debug_id, report = False, block = False):
             time.sleep(50./1000)
     else:
         raise Exception('failed to attach')
-    attach_process_from_socket(conn, report, block)
+    attach_process_from_socket(conn, debug_options, report, block)
+
+def attach_process_from_socket(sock, debug_options, report = False, block = False):
+    global conn, attach_sent_break, DETACHED, DEBUG_STDLIB, BREAK_ON_SYSTEMEXIT_ZERO, DJANGO_DEBUG
+
+    BREAK_ON_SYSTEMEXIT_ZERO = 'BreakOnSystemExitZero' in debug_options
+    DJANGO_DEBUG = 'DjangoDebugging' in debug_options
+
+    if '' in PREFIXES:
+        # If one or more of the prefixes are empty, we can't reliably distinguish stdlib
+        # from user code, so override stdlib-only mode and allow to debug everything.
+        DEBUG_STDLIB = True
+    else:
+        DEBUG_STDLIB = 'DebugStdLib' in debug_options
+
+    def _excepthook(exc_type, exc_value, exc_tb):
+        # Display the exception and wait on exit
+        if exc_type is SystemExit:
+            if ('WaitOnAbnormalExit' in debug_options and exc_value.code != 0) or ('WaitOnNormalExit' in debug_options and exc_value.code == 0):

```

Comment:
It's probably fine, but I think I'd rather capture the result of `'WaitOn(Abn|N)ormalExit' in debug_options` outside this function, just in case `debug_options` changes/disappears during execution.